lib/syskit/data_service.rb

```diff
@@ -90,10 +90,13 @@ def find_device_attached_to(service = nil)
                     elsif driver_services.size > 1
                         raise ArgumentError, "#{self} handles more than one device, you must specify one of #{driver_services.map(&:name).sort.join(", ")} explicitely"
                     end
-                    service = driver_services.first
+                    service = driver_services.find do |driver_service|
+                        service == driver_service
+                    end
+                    
                 end
 
-                arguments["#{service.name}_dev"]
```

Comment:
> This change implies that in a concrete component network each concrete TaskContext can only be driver for one particular device. I think this is a good implication and also mirrors the behavior from before this PR (where a task context could only be driver for one device type).

It's not a good implication since we had (and probably will have again) drivers that could drive multiple devices at the same time. I originally made the feature of having multiple drivers declared per task for this particular purpose.

I don't understand the comment about the behaviour before the PR. The behaviour should be that a task context can be driver for an arbitrary number of devices of any types. The argument name is unique on a per-bound service (i.e. it is using the as: argument of driver_for).